### ISSUE TITLE
Speeding up builds using a cached Docker image

### ISSUE THREAD

**Issue by sgrodzicki** [mrsked/mrsk#295]
While running MRSK from a CI server (e.g., Cloud Build) it would great if one could use a cached Docker image as described in [Google's best practices to speed up builds](https://cloud.google.com/build/docs/optimize-builds/speeding-up-builds#using_a_cached_docker_image).

Before:
```
docker build -t [IMAGE_NAME]:latest &&
docker push [IMAGE_NAME]:latest
```

After:
```
docker pull [IMAGE_NAME]:latest &&
docker build -t [IMAGE_NAME]:latest --cache-from [IMAGE_NAME]:latest &&
docker push [IMAGE_NAME]:latest
```

**Comment by jeremy** [comment]
References #159

**Comment by sgrodzicki** [comment]
I've patched `main` with changes from #159 and added `docker pull` in https://github.com/sgrodzicki/mrsk/commit/f3c8debd2d1cfbaafcff526c7cec219f2313544d. In my case the deploy step went down from 3m 32s to 1m 9s (-67%). Do you want to revisit adding that change?
